scipy/stats/_stats_py.py

```diff
@@ -617,14 +621,18 @@ def tmean(a, limits=None, inclusive=(True, True), axis=None):
     10.0
 
     """
-    a = asarray(a)
+    if not np.issubdtype(a.dtype, np.inexact):
```

Comment:
To confirm, this is so you can put NaNs in it, right?

Reply:
Yes.